content/articles/run-length-encoding/index.md

```diff
@@ -0,0 +1,150 @@
+
+
+For the lossy algorithms, we use it where quality compromising is acceptable. We achieve high compression ratios hence greater size reduction.
+The lossy compression process is non-reversible unlike the one for lossless compression algorithms.
+Examples include Lossy predictive coding,  Block transform coding, and Vector quantization. (The links to these types of coding are found at the end of this article)
+
+### Run Length Encoding(RLE).
+
+This article is dedicated to RLE and its implementation using a simple Python code.
+
+RLE is a statistical coding scheme where data is encoded based on their probability of occurrence. 
+
+RLE reduces repetitive, consecutive data called **runs**. It does so by storing the number of these runs followed by the data. Let's look at examples.
+
+1. For the text `AAAAAAAAAAAAAHHHEEM`(*19 characters*), RLE will encode it to get this `13A3H2EM`(7 characters).
+
+2. For the text `AAAAHHHEEM, HAHA.`(*17 characters*), it will be encoded as `4A3H2E1M1,1 1H1A1H1A1.`(*22 characters*).
+
+From these examples, we see that RLE is suitable for compressing data with very many runs but not a few runs e.g image pixel data. This is because short runs may end up taking the same space or more as we see in the second example.
+
+#### The Python code.
+
+Let's jump into the Python code.
+
+A basic understanding of Python is required. Familiarity with a notebook-based interface like Google Colab is required because we will be using that. You can as well use it in normal Python files.
+
+It's a very simple algorithm to implement.
+
+Create a new notebook and add the following code in the first cell.
+
+```python
+def encode_message(message):
+    encoded_string = ""
+    i = 0
+    while (i <= len(message)-1):
+        count = 1
+        ch = message[i]
+        j = i
+        while (j < len(message)-1): 
+            if (message[j] == message[j + 1]): 
+                count = count + 1
+                j = j + 1
+            else: 
+                break
+        encoded_string = encoded_string + str(count) + ch
+        i = j + 1
+    return encoded_string
+```
+
+The code is for encoding a message.
+It is a function `encode_message()` with a parameter `message` for the string to be encoded.
+
+The variable, `encoded_string`, is for storing the encoded string. The `while` loop initializes the count to 1 for storing the human-readable count, 1, 2, 3, 4,...., instead of starting from 0. 
+
+The innermost `while` loop checks if the character at the current index is the same as the character at the next index. If the characters are the same, the count is incremented to 1. If they are not the same, the counter is stopped, then the count and the character is concatenated to the `encoded_string` variable. This goes on until all the characters are encoded.
+
+The `encoded_string` is returned after the function has finished executing.
+
+The next function is for decoding the encoded message. Create a new cell and paste this code.
```

Comment:
`Create a new cell`

I guess you are using either Jupyter notebook or Google Colab. Make sure to specify it before hand, in the `Prerequisites` section.